### ISSUE TITLE
Investigate possibility of adding alt-text field for image widgets

### ISSUE THREAD

**Issue by caaster** [sul-dlss/exhibits#819]
For @ggeisler -- per our discussion re accessibility & also ability for SUL reference to help patrons identify unlabeled exhibit images

**Comment by ggeisler** [comment]
This issue is trickier than it might seem on the surface. The biggest complication is that the determination of whether the `alt` attribute for an image should have a value or be left empty depends on the surrounding context. You typically don't want to provide an `alt` description for an image if the surrounding context already conveys what the image conveys. (You also don't want to provide an `alt` description for purely decorative images, but that shouldn't apply to widget-based images.)

We could address this by providing an input field for the curator who is configuring a widget to provide (or intelligently not provide, given the context of the widget image) a sensible, original `alt` description for each item added to a widget. But I don't think that approach is practical, because it is unlikely many curators will take the time to add an `alt` description or know when it is appropriate to do so. 

An automated solution is more practical in that we can reliably provide an `alt` description for every image. In some cases the description could be redundant with the surrounding context, which is not ideal for someone with a screenreader, but if we assume most of the time the `alt` description has value, it could be a worthwhile tradeoff.

So how do we get the text to use for each image's `alt` description, if we're doing it in an automated way? The abstract field potentially contains the most useful text for conveying what the image is about but: 1) the abstract can be very long and 2) based on a random sampling of images used in current exhibits, most objects don't have an abstract field value.

All objects should have a title value, however, and the title of many objects is at least somewhat descriptive of the image. So my current thought is that the solution to this issue is, for items that are used in exhibit widgets, we use the title of the item to populate the `alt` attribute when the widget is rendered. I'm not sure whether there are any technical hurdles to doing this, but since we know the title of the item when it is configured in the widget, I'm assuming we can get that value when rendering the configured widget.

That's my preliminary analysis, but I'm going to leave this as Design Needed until I can discuss with Jennifer and think about it a bit more.

**Comment by corylown** [comment]
Closing in favor of #2462
